sys/include/net/ng_udp.h

```diff
@@ -53,6 +53,13 @@ extern "C" {
 #endif
 
 /**
+ * @brief   The PID to the UDP thread.
+ *
+ * @note    Use @ref ng_udp_init() to initialize. **Do not set by hand**.
+ */
+extern kernel_pid_t ng_udp_pid;
```

Comment:
In #3296 @haukepetersen wrote

> I don't like this change... Feels wrong to expose more internals just for testing. Exposing this leads potentially to people miss-using this... Why don't you just use netreg in the test's for lookup?

and @kaspar030 seconded it. 

My answer:
To me it feels wrong to trust in the tests, that the PID set in `netreg` is really the PID I am looking for. Anyone can register as UDP handler so from a black-box tester perspective it seems not right to use `netreg` (even though I know as an implementater that nothing else is registered). I could loop over all registered PID but that would greatly change the tests and make them more complicated. The misuse you spoke about can be easily prevented with proper doc (which I did here) and review. Considering my concerns have you a better idea?